### PR TITLE
fix(clerk-react): Add data-clerk-publishable-key attribute only when …

### PR DESCRIPTION
**Pull request by nikosdouvlis**
…PK is available

## Type of change

- [ ] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

## Packages affected

- [ ] `@clerk/clerk-js`
- [ ] `@clerk/clerk-react`
- [ ] `@clerk/nextjs`
- [ ] `@clerk/remix`
- [ ] `@clerk/types`
- [ ] `@clerk/themes`
- [ ] `@clerk/localizations`
- [ ] `@clerk/clerk-expo`
- [ ] `@clerk/backend`
- [ ] `@clerk/backend-core`
- [ ] `@clerk/clerk-sdk-node`
- [ ] `@clerk/edge`
- [ ] `@clerk/shared`
- [ ] `build/tooling/chore`

## Description

- [ ] `npm test` runs as expected.
- [ ] `npm run build` runs as expected.

### CODE REVIEW

File: packages/react/src/types.ts [hunk 1]
```diff
@@ -21,16 +21,10 @@ export type IsomorphicClerkOptions = ClerkOptions & {
   clerkJSVariant?: 'headless' | '';
 } & (
     | {
-        /**
-         * @deprecated Use `publishableKey` instead
```
Comment (nikosdouvlis) [inline packages/react/src/types.ts:25]: Removed the deprecation warning as Kima is not enabled yet

File: packages/react/src/utils/scriptLoader.ts [hunk 2]
```diff
@@ -83,8 +78,12 @@ export async function loadScript(params: LoadScriptParams): Promise<HTMLScriptEl
...
 
     const script = document.createElement('script');
-    script.setAttribute('data-clerk-publishable-key', publishableKey!);
-    script.setAttribute('data-clerk-frontend-api', frontendApi!);
+    if (publishableKey) {
+      script.setAttribute('data-clerk-publishable-key', publishableKey);
+    }
+    if (frontendApi) {
+      script.setAttribute('data-clerk-frontend-api', frontendApi);
+    }
```
Comment (SokratisVidros) [inline packages/react/src/utils/scriptLoader.ts:86]: 🔧 Let's throw on error if non of them is set at this point via the errorThrower.

**Review by SokratisVidros — APPROVED** [review]

**Review by colinclerk — APPROVED** [review]

File: packages/react/src/utils/scriptLoader.ts (hunk 2 above)
Comment (nikosdouvlis) [reply packages/react/src/utils/scriptLoader.ts:86]: Thought of that as well, but it felt that this is too deep for an error as we already check a couple of times. Im going to add one as per your suggestion, better be on the safe side

Comment (nikosdouvlis) [reply packages/react/src/utils/scriptLoader.ts:90]: Done

**Review by agis — APPROVED** [review]

**Pull request merged**